lib/core/src/Cardano/Wallet.hs

```diff
@@ -1624,26 +1633,34 @@ signTx
         , HardDerivation k
         , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
         , WalletKey k
+        , input ~ (TxIn, TxOut)
+        , output ~ TxOut
         )
     => ctx
     -> WalletId
     -> Passphrase "raw"
     -> Maybe TxMetadata
-    -> UnsignedTx (TxIn, TxOut)
+    -- This function is currently only used in contexts where all change outputs
+    -- have been assigned with addresses and are included in the set of ordinary
+    -- outputs. We use the 'Void' type here to prevent callers from accidentally
+    -- passing change values into this function:
+    -> UnsignedTx input output Void
```

Comment:
for quite a while have not seen Void :-)

Reply:
Originally I wrote this by specifying `change ~ Void` in the list of constraints, and then specifying `UnsignedTx input output change` as the parameter type, as I felt that this would be more readable.

However, GHC views this as a redundant constraint. Of course, it's possible to silence the redundant constraint warning, but that would mean silencing this warning for the whole file. So I eventually switched to just writing `Void` inline, even though it is slightly less readable (IMO).